src/omnicore/tx.cpp

```diff
@@ -82,6 +82,16 @@ bool CMPTransaction::isOverrun(const char* p)
 
 // -------------------- PACKET PARSING -----------------------
 
+/** Temporarily disables packet debugging and then calls interpret_Transaction(). */
+bool CMPTransaction::reinterpret_Transaction()
+{
+    bool packetDebugState = msc_debug_parser;
+    msc_debug_parser = 0;
```

Comment:
If you step back a moment, and forget the current architecture. How would the ideal logging and code flow look like? (sort of a rhetorical question to fuel some brainstorming)

I mean, the middle ground could be to add a flag to `CMPTransaction`, which indicates, whether it's RPC only or not.